### PR TITLE
Implement hierarchical aggregation for Tabulator

### PR DESCRIPTION
**Pull request by philippjfr**
Screen Shot 2021-08-09 at 11 12 21 AM

Fixes https://github.com/holoviz/panel/issues/2619

### CODE REVIEW

File: panel/widgets/tables.py [hunk 1]
```diff
@@ -162,7 +172,7 @@ def _get_column_definitions(self, col_names, df):
                 col_kwargs['width'] = 0
 
             title = self.titles.get(col, str(col))
-            if col in indexes and len(indexes) > 1 and self.hierarchical:
+            if col in indexes and len(indexes) > 1 and getattr(self, 'hierarchical', False):
```
Comment (philippjfr) [inline panel/widgets/tables.py:175]:
```suggestion
            if col in indexes and len(indexes) > 1 and self.hierarchical:
```

File: panel/models/tabulator.ts [hunk 2]
```diff
@@ -250,7 +291,10 @@ export class DataTabulatorView extends PanelHTMLBoxView {
           tab_column.formatter = "tickCross"
         else {
           tab_column.formatter = (cell: any) => {
-            return column.formatter.doFormat(cell.getRow(), cell, cell.getValue(), null, null)
+            const formatted = column.formatter.doFormat(cell.getRow(), cell, cell.getValue(), null, null)
+	    const node = div()
+	    node.innerHTML = formatted
+	    return node.children[0].innerHTML
```
Comment (mathrick) [inline panel/models/tabulator.ts]: This completely breaks `HTMLTemplateFormatter`:

```python
pn.widgets.Tabulator(pd.DataFrame({'link': ['foo']}), 
                     formatters={'link': HTMLTemplateFormatter(
                         template='<a href="<%= value %>">Link</a>'
                     )})
```

Comment (philippjfr) [reply panel/models/tabulator.ts]: Thanks, can you open an issue?

Comment (mathrick) [reply panel/models/tabulator.ts]: Done, #2730

**Pull request merged**